src/e2e/java/uk/gov/hmcts/reform/tools/builders/Appeal.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.hmcts.reform.tools.builders;
+
+import java.io.IOException;
+import java.util.*;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import uk.gov.hmcts.reform.tools.utils.*;
+
+public class Appeal {
+
+    public Date lastUpdatedDate = new Date();
+
+    public AppealHeader header = new AppealHeader();
+    public AppealParties parties = new AppealParties();
+    public AppealMajorStatus majorStatus = new AppealMajorStatus();
+    public AppealHearing hearing = new AppealHearing();
+
+    private final XmlWriter xmlWriter;
+    private Map<String, String> appealHeaderStore = new HashMap<>();
+    private final List<Map<String, String>> appealPartiesStore = new ArrayList<>();
+    private final List<Map<String, String>> appealMajorStatusList = new ArrayList<>();
+    private final List<Map<String, String>> appealHearingStore = new ArrayList<>();
+    private Integer daysOffSet = 0;
+
+    public Appeal(XmlWriter xmlWriter) throws ParserConfigurationException {
+        this.xmlWriter = xmlWriter;
+    }
+
+    public void setUpdatedDaysAGo(Integer offset) {
+        lastUpdatedDate = TestContainer.backDate(new Date(), offset);
+    }
+
+    public void setStartedDaysAGo(Integer offSet) {
+        daysOffSet = offSet;
+        TestContainer.backDate(lastUpdatedDate, offSet);
+    }
+
+    public String getLastUpdatedDate() {
+        return TestContainer.asGapsDate(lastUpdatedDate);
+    }
+
+    public String getFutureDate(Integer offSet) {
+        daysOffSet = offSet;
+        Date futureDate = TestContainer.backDate(lastUpdatedDate, -offSet);
+        return TestContainer.asGapsDate(futureDate);
+    }
+
+    public String getCaseId() {
+        return header.getAppealCaseId();
+    }
+
+    public String getCaseRefNum() {
+        return header.getAppealCaseRefNum();
+    }
+
+
+    public String getCaseRefNumber() {
+        return header.getAppealCaseRefNum();
+    }
+
+    public String onDay(Integer daysAdded) {
+        Date date = TestContainer.backDate(lastUpdatedDate, daysOffSet - daysAdded);
+        return TestContainer.asGapsDate(date);
+    }
+
+    public Appeal storeHeader() {
+        appealHeaderStore = header.build();
+        return this;
+    }
+
+    public Appeal storeParties() {
```

Comment:
I find these methods really confusing to read. Why are you constructing a new `parties` object after building it? Maybe i'm missing something but I think some of the methods in this class can be rewritten to be abit cleaner